### ISSUE TITLE
Default log cannot record remote ip address when nginx proxy pass to unix address

### ISSUE THREAD

**Issue by sunshineplan** [gin-gonic/gin#2962]
I create my app and serve it on an unix address.
```go
package main

import (
	"log"
	"net"
	"net/http"

	"github.com/gin-gonic/gin"
)

func main() {
	listener, err := net.Listen("unix", "unix path")
	if err != nil {
		log.Fatal(err)
	}
	server := new(http.Server)
	router := gin.Default()
	router.GET("/", func(c *gin.Context) { c.String(200, c.Request.RemoteAddr) })
	server.Handler = router
	server.Serve(listener)
}

```
nginx setting:
```
location / {
   include proxy_params;
   proxy_pass http://unix:unix path:/;
}
```
After doing this, I always get the http request with RemoteAddr set as "@", which cannot record by default logger, because it will failed in net.SplitHostPort function.

**Comment by h1zzz** [comment]
I think you may need to add the following configuration to your nginx server. I haven't tried it, Please let me know if it is useful.
```
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```

**Comment by Bisstocuz** [comment]
If you didn't use any proxy at front of `nginx`, there is a possible temporary solution:
1. Add this line to `nginx` config file:
```
proxy_set_header X-Client-IP $remote_addr;
```
2. Set this in your Go project code:
```go
	router.TrustedPlatform = "X-Client-IP"
```

**Comment by sunshineplan** [comment]
> I think you may need to add the following configuration to your nginx server. I haven't tried it, Please let me know if it is useful.
> 
> ```
> proxy_set_header X-Real-IP $remote_addr;
> proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
> ```

@h1zzz, I have already added "include proxy_params" which include these lines, so it will not be a solution.

**Comment by sunshineplan** [comment]
> If you didn't use any proxy at front of `nginx`, there is a possible temporary solution:
> 
> 1. Add this line to `nginx` config file:
> 
> ```
> proxy_set_header X-Client-IP $remote_addr;
> ```
> 
> 2. Set this in your Go project code:
> 
> ```go
> 	router.TrustedPlatform = "X-Client-IP"
> ```

Thanks @Bisstocuz, it works for me, and I think it's the most effective way. But I must change all my apps' code as well as theirs nginx config.
I can't find any refers about why remote address will be set as single "@", shall we trust it in package?

**Comment by Bisstocuz** [comment]
> Thanks @Bisstocuz, it works for me, and I think it's the most effective way. But I must change all my apps' code as well as theirs nginx config. I can't find any refers about why remote address will be set as single "@", shall we trust it in package?

Yep, maybe we need open an issue in `github.com/golang/go` for asking.

**Comment by sunshineplan** [comment]
According #https://github.com/golang/go/issues/49825
```
"Abstract" Unix domain socket. Rewrite leading NUL as @ for textual display. (This is the standard convention.) 
Not friendly to overwrite in place, but the callers below don't care.
```

I also found this problem will happen on Windows 10 and Linux (I tested on Debian 11), but not on Mac (RemoteAddr will be empty string). So my PR is not a good idea to solve this problem, I'll close it.
I decided simply set router.TrustedPlatform as X-Real-IP, after that I do not need to change my nginx config.
